### PR TITLE
Update Router unmount to check if other versions exist

### PR DESCRIPTION
**Pull request by gcssabbagh**
Updated Router unmount() to check if there are other versions for a certain route to fix https://github.com/restify/node-restify/issues/1171

First open-source contribution here ever.. so hope I'm doing it right.

I used the existing Router findByPath method instead of an O(n) filter()

Cheers,
Georges

### CODE REVIEW

File: test/server.test.js [hunk 1]
```diff
@@ -229,6 +229,33 @@ test('rm route and clear cached route', function (t) {
     });
 });
 
+test('rm one version of the routes, other versions should still work',
```
Comment (DonutEspresso) [inline test/server.test.js]: Can you prepend this test name with the issue it is fixing? 'GH-1171: rm one version...'

File: test/server.test.js [hunk 2]
```diff
@@ -229,6 +229,33 @@ test('rm route and clear cached route', function (t) {
...
+
+        var routeThree = SERVER.get({ path: '/hello/:name', version: '3.0.0'},
+            function (req, res, next) {
+                res.send('hello ' + req.params.name);
+                next();
+            });
+
+        t.ok(SERVER.rm(routeThree));
+
+        CLIENT.get('/hello/whatever', function (err, _, res) {
```
Comment (DonutEspresso) [inline test/server.test.js]: Can we do a strict check here, and make a request specifically for both 1.0.0 and 2.0.0 to ensure they're both available? Should be able to add a header version to do that.

**Comment by DonutEspresso** [comment]
Thanks for the PR, and congrats on your first contribution! :tada: 

I left some minor comments. I think it would also be helpful to add another test that ensures a 404 after removing all the routes, since that's functionality that's now gated behind the new check you added.

**Comment by gcssabbagh** [comment]
Thanks @DonutEspresso for the feedback! I'm happy to contribute, loving Restify so far :)

I have added tests to look at specific versions (200 on the remaining versions, 400 on the removed version) and a test for 404 once all route versions are removed as you specified

**Comment by DonutEspresso** [comment]
Cheers - this LGTM. 

@yunong can I get a another pair of eyes? Just in case I missed something.

**Comment by yunong** [comment]
LGTM

**Pull request merged**